File: atlas-sdk/JavaScript/entity.js
class Entity {
  constructor(id, name, type = "default") {
    this.id = id;
    this.name = name;
    this.type = type;
    this.metadata = {};
    this.createdAt = Date.now();
  }

  setMeta(key, value) {
    this.metadata[key] = value;
  }

  getMeta(key) {
    return this.metadata[key];
  }

  toDict() {
    return {
      id: this.id,
      name: this.name,
      type: this.type,
      metadata: this.metadata,
      created_at: this.createdAt,
    };
  }

  static fromDict(data) {
    const e = new Entity(data.id, data.name, data.type || "default");
    e.metadata = data.metadata || {};
    e.createdAt = data.created_at || Date.now();
    return e;
  }
}

module.exports = { Entity };

File: atlas-sdk/Go/entity.go
package atlas

import "time"

type Entity struct {
	ID        string                 `json:"id"`
	Name      string                 `json:"name"`
	Type      string                 `json:"type"`
	Metadata  map[string]interface{} `json:"metadata"`
	CreatedAt int64                  `json:"created_at"`
}

func NewEntity(id, name, entityType string) *Entity {
	return &Entity{
		ID:        id,
		Name:      name,
		Type:      entityType,
		Metadata:  make(map[string]interface{}),
		CreatedAt: time.Now().UnixMilli(),
	}
}

func (e *Entity) SetMeta(key string, value interface{}) {
	e.Metadata[key] = value
}

func (e *Entity) GetMeta(key string) interface{} {
	return e.Metadata[key]
}
